### PR TITLE
validator(dm): fast mode validation

### LINKED ISSUE #4716: validator(dm): fast validation mode

### PR DESCRIPTION
**Pull request by buchuitoudegou**
### What problem does this PR solve?

Issue Number: close #4716 

### What is changed and how it works?
1. extract mode config from subtask config
2. only validate PK/UK

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

- #4539 don't merge until this PR finished

### Release note 

```release-note
`None`
```

### CODE REVIEW

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: dm/syncer/data_validator.go [hunk 1]
```diff
@@ -120,6 +120,8 @@ type DataValidator struct {
 
 	// such as table without primary key
 	unsupportedTable map[string]string
+	waitSyncerTimer  *time.Timer
```
[the remark this answers is not in the document]
Comment (buchuitoudegou) [reply dm/syncer/data_validator.go:123]: It might be introduced by merging the latest branch by mistake😂. I'll fix it.

**Review by lance6716 — APPROVED** [review]

File: dm/dm/worker/subtask.go [hunk 2]
```diff
@@ -268,6 +268,8 @@ func (st *SubTask) StartValidator(expect pb.Stage) {
 	}
 	st.Lock()
 	defer st.Unlock()
+	// refetch the validation config
```
Comment (D3Hunter) [inline dm/dm/worker/subtask.go:271]: what's this comments for? subtask config is fetched in here
https://github.com/pingcap/tiflow/blob/0a4fc06efc74bc5afdc81d20e6bc872946b597ef/dm/dm/worker/source_worker.go#L1276-L1277

might need to update cfg of validator to update validation mode, we can do this later.

File: dm/syncer/data_validator.go [hunk 3]
```diff
@@ -120,6 +120,7 @@ type DataValidator struct {
 
 	// such as table without primary key
 	unsupportedTable map[string]string
+	mode             string
```
Comment (D3Hunter) [inline dm/syncer/data_validator.go:123]: cfg.ValidationConfig already has this field(validator worker has this cfg too), no need to maintain this field separately.

File: dm/dm/worker/subtask.go (hunk 2 above)
Comment (buchuitoudegou) [reply dm/dm/worker/subtask.go:271]: Merging mistake😂 I'll fix it

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**